pkg/imgpkg/bundle/images_lock.go

```diff
@@ -13,52 +13,94 @@ import (
 
 //go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
 
+type ImageRef struct {
+	lockconfig.ImageRef
+	IsBundle bool
+}
+type ImagesRef []ImageRef
+
+func (i *ImagesRef) AddImagesRef(refs ...ImageRef) {
+	result := *i
+	for _, ref := range refs {
+		found := false
+		for j, imageRef := range result {
+			if imageRef.Image == ref.Image {
+				found = true
+				result[j] = ref
+				break
+			}
+		}
+
+		if !found {
+			result = append(result, ref)
```

Comment:
this is getting a bit funky with reassigning i pointer. to make it a bit less fragile, lets switch to

```
type ImageRefs struct {
  refs []ImageRef
}
```

Reply:
I did this, the only funky thing is that now you need to call `ImageRefs.ImageRefs` to get the refs instead of just iterating over the `ImagesRefs` Slice